SalesforceCommerceCloud/pwa-kit
# Filter link on search displays results

# Description

Currently, when users search for an item and then apply a filter from the search results page, they are directed to a 404 error page that looks like this: 
https://pwa-kit.mobify-storefront.com/en-GB/category/undefined?limit=25&q=dress&refine=c_refinementColor%3DBlack&sort=best-matches 

This fix ensures that users can filter on search results.


# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- If users search for an item, "categoryId" is undefined. If "categoryId" is undefined, the url is correctly built with "/search?" and the selected refinement categories

# How to Test-Drive This PR

- Run `npm start`
- Go to http://localhost:3000/
- Search "dress"
- Click on the color filter "Black"
- See that the filtered results are displayed on the page and that the url is equal to: http://localhost:3000/global/en-GB/search?limit=25&q=dress&refine=c_refinementColor%3DBlack&sort=best-matches

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
